ampd/src/commands/bond_worker.rs

```diff
@@ -0,0 +1,48 @@
+use std::path::Path;
+
+use cosmrs::cosmwasm::MsgExecuteContract;
+use cosmrs::tx::Msg;
+use cosmrs::Coin;
+use error_stack::Result;
+use report::ResultCompatExt;
+use service_registry::msg::ExecuteMsg;
+use valuable::Valuable;
+
+use crate::commands::{broadcast_tx, worker_pub_key};
+use crate::config::Config;
+use crate::{Error, PREFIX};
+
+#[derive(clap::Args, Debug, Valuable)]
+pub struct Args {
+    pub amount: u128,
+    pub denom: String,
+}
+
+pub async fn run(config: Config, state_path: &Path, args: Args) -> Result<(), Error> {
+    let coin = Coin::new(args.amount, args.denom.as_str()).change_context(Error::Command)?;
```

Comment:
be more specific about why this fails. Seems weird that only the coin parsing fails with "command failed", while the rest shows some other error.